compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.td

```diff
@@ -456,4 +456,44 @@ def IREEGPU_YieldOp : Op<IREEGPU_Dialect, "yield", [
       [{  attr-dict ($values^ `:` type($values))? }];
 }
 
+//===----------------------------------------------------------------------===//
+//
+// AMD Specific Operations
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// BufferResourceCastOp
+//===----------------------------------------------------------------------===//
+
+def IREEGPU_BufferResourceCastOp : Op<IREEGPU_Dialect, "buffer_resource_cast", [
+  Pure,
+  AllTypesMatch<["input", "result"]>]> {
+  let summary = "Represents a cast to addr_space<7> (buffer resource) before bufferization";
+  let description = [{
+    Nominal cast of a tensor to AMDGPU buffer resource memory space before
+    bufferization. This op takes the parameters with which to perform the cast
+    if |input| bufferizes to `storage_buffer` memory space. If |input| resolves
+    to any other memory space this op is silently dropped and has no effect.
+
+    If |cache_swizzle_stride| is present, there is verification before
```

Comment:
Why're you using `|param|`? Is that a style thing I'm unaware of?

Reply:
It's pretty common through the rest of the codebase, e.g.

- https://github.com/iree-org/iree/blob/72f11570a2582f2e1703dc00eda6780fcb8f8c5d/runtime/src/iree/vm/buffer.h#L76-L82
- https://github.com/iree-org/iree/blob/72f11570a2582f2e1703dc00eda6780fcb8f8c5d/compiler/src/iree/compiler/Dialect/Util/IR/UtilOps.td#L345

but isn't formal style as far as I am aware, I just like it over backticks or quotes.